astroid/as_string.py

```diff
@@ -15,9 +15,33 @@
 """
 import sys
 
+from astroid import node_classes
```

Comment:
@brycepg mentioned how this improvement might work without importing `node_classes` here, by providing a precedence method for the base `NodeNG` class which can then be implemented by the rest of nodes that you are interested in.

But now I see that we also use `node_classes` for other use cases as well, such as to see if the first node from `orelse` is an `If` node. Instead of doing a type check against the class itself, we prefer doing a `.__class__.__name__` check instead. There's a third option to get past this cyclical dependency, which is to use the `lazy_import` function from `astroid.util`, but let's see if the first two solutions work before that.